### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by mturilli**
This is the first pass to our documentation. Not complete yet but will serve as a base for future contributions.

### CODE REVIEW

File: docs/source/components.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
+.. _chapter_components:
+
+===============
+Core Components
+===============
+
+The SDK has four core components but it is open to the contribution of any system that support the execution of scientific workflows on the Departiment of Energy high performance computing platforms.
```
Comment (jameshcorbett) [inline docs/source/components.rst:7]: Typo: departiment -> department

File: docs/source/components.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+*  `Flux <http://flux-framework.org>`_. Workload management system (à la Slurm, PBS, LSF), with single-user and multi-user (a.k.a. system instance) modes.
+*  `Parsl <https://parsl-project.org>`_. Pure Python library for describing and managing parallel computation on one to many nodes.  It contains abstractions to support various parallel and distributed workloads: from bag-of-tasks to dataflow, short to long duration tasks, single core through to multi-node.
+*  `PSI/J <https://github.com/ExaWorks/psi-j-python>`_. The Portable Submission Interface for Jobs TODO: add description here.
```
Comment (jameshcorbett) [inline docs/source/components.rst:11]: a Python library providing a single interface to a variety of HPC resource managers, including LSF, Flux, Slurm, Cobalt, and PBS.

**Review by jameshcorbett — APPROVED** [review]
Looks like a great start to me. I can see there are some missing pieces but I think it's an improvement and so should go in.

File: docs/source/components.rst (hunk 2 above)
Comment (jameshcorbett) [reply docs/source/components.rst:11]: (That's a suggestion for eliminating the "TODO:")

File: docs/source/components.rst (hunk 1 above)
Comment (mturilli) [reply docs/source/components.rst:7]: Thanks.

File: docs/source/components.rst (hunk 2 above)
Comment (mturilli) [reply docs/source/components.rst:11]: Link updated and added a short description.

**Pull request merged**